composition-js/src/merging/merge.ts

```diff
@@ -2003,7 +2034,7 @@ class Merger {
     for (const source of sources) {
       if (source) {
         for (const directive of source.appliedDirectives) {
-          if (this.isMergedDirective(directive)) {
+          if (this.isMergedDirective(directive) || this.exposedDirectives().includes(directive.name)) {
```

Comment:
It would have made more sense to me to move the "is exposed" check inside `isMergedirective` (at least that was the original intend of that `isMergedDirective` method: to capture the full extend of when a directive was merged or not). Or was there a specific reason not too?

Reply:
The reason that it's outside of it is because `isMergedDirective` returns true if the directive is executable, regardless of whether or not we want to expose it. I'll see what I can do to refactor.

Reply:
> The reason that it's outside of it is because `isMergedDirective` returns true if the directive is executable, regardless of whether or not we want to expose it

But juste to make sure we're on the same page, I don't think we want to change that anyway. That is, `isMergedDirective` returns true for directive _definition_ having executive locations, but that should still be true regardless of what is exposed.

Looking at it, I believe just adding:
```typescript
if (this.exposedDirectives().includes(definition.name)) {
  return true;
}
```
at the very beginning of `isMergedDirective` should be the correct thing to do. But if there is a reason this doesn't work I'm not seeing, it might be worth some commenting.